src/GDBDebugSession.ts

```diff
@@ -861,10 +861,12 @@ export class GDBDebugSession extends LoggingDebugSession {
     ): Promise<void> {
         try {
             if (!this.isRunning) {
-                const result = await mi.sendThreadInfoRequest(this.gdb, {});
-                this.threads = result.threads.map((thread) =>
-                    this.convertThread(thread)
-                );
+                if (!this.isRunning) {
+                    const result = await mi.sendThreadInfoRequest(this.gdb, {});
+                    this.threads = result.threads.map((thread) =>
+                        this.convertThread(thread)
+                    ).sort((a, b) => a.id - b.id);
+                }
```

Comment:
Looks like you have a small error with a double-check of `this.isRunning`
```suggestion
                const result = await mi.sendThreadInfoRequest(this.gdb, {});
                this.threads = result.threads.map((thread) =>
                    this.convertThread(thread)
                ).sort((a, b) => a.id - b.id);
```

Reply:
Hi @jonahgraham ,

I changed some code based on your suggestion, and with this change, it worked fine.
But when I commit the code, I get the build fails error, apparently due to a server problem. 
Could you please help me review it? And also if this code is merged, can you please build the latest version? we need it to test for an upcoming release.

Thank you so much!

Reply:
You seem to have not applied my suggestion properly. In commit [17f6704](https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter/pull/267/commits/17f67042eef237cfd2676644f000f92b2a55a1c3) you applied my suggestion, but then in the subsequent commit you went further and removed `if (!this.isRunning) {`

This was the original comment:

> Looks like you have a small error with a double-check of this.isRunning

It was because you had this as the code:

```typescript
            if (!this.isRunning) {
                if (!this.isRunning) {
                    const result = await mi.sendThreadInfoRequest(this.gdb, {});
                    this.threads = result.threads.map((thread) =>
                        this.convertThread(thread)
                    ).sort((a, b) => a.id - b.id);
                }
            }
```

The tests are probably failing because you are trying to call thread info while target is running.